storagemarket/lp2pimpl/net.go

```diff
@@ -196,6 +199,13 @@ func (p *DealProvider) Start(ctx context.Context) {
 	p.host.SetStreamHandler(DealProtocolv120ID, p.handleNewDealStream)
 
 	p.host.SetStreamHandler(DealStatusV12ProtocolID, p.handleNewDealStatusStream)
+
+	// Handle legacy deal stream here and reject all legacy deals
+	if !p.prov.IsEnabledLegacyStorageDeals() {
```

Comment:
Instead of adding this flag to storagemarket.Provider, I'd suggest adding it to DealProvider (and passing it to NewDealProvider as a parameter)